graylog2-server/src/main/java/org/graylog2/periodical/IndexSetsMigrationPeriodical.java

```diff
@@ -0,0 +1,165 @@
+    private final ClusterConfigService clusterConfigService;
+
+    @Inject
+    public IndexSetsMigrationPeriodical(final ElasticsearchConfiguration elasticsearchConfiguration,
+                                        final Map<String, Provider<RotationStrategy>> rotationStrategies,
+                                        final Map<String, Provider<RetentionStrategy>> retentionStrategies,
+                                        final IndexSetService indexSetService,
+                                        final ClusterConfigService clusterConfigService) {
+        this.elasticsearchConfiguration = elasticsearchConfiguration;
+        this.rotationStrategies = rotationStrategies;
+        this.retentionStrategies = retentionStrategies;
+        this.indexSetService = indexSetService;
+        this.clusterConfigService = clusterConfigService;
+    }
+
+    @Override
+    public void doRun() {
+        final IndexManagementConfig indexManagementConfig = clusterConfigService.get(IndexManagementConfig.class);
+        checkState(indexManagementConfig != null, "Couldn't find index management configuration");
+
+        final RotationStrategyConfig rotationStrategyConfig = getRotationStrategyConfig(indexManagementConfig);
+        final RetentionStrategyConfig retentionStrategyConfig = getRetentionStrategyConfig(indexManagementConfig);
+
+        final IndexSetConfig indexSetConfig = IndexSetConfig.builder()
+                .title("Default index set")
+                .description("The Graylog default index set")
+                .indexPrefix(elasticsearchConfiguration.getIndexPrefix())
+                .shards(elasticsearchConfiguration.getShards())
+                .replicas(elasticsearchConfiguration.getReplicas())
+                .rotationStrategy(rotationStrategyConfig)
+                .retentionStrategy(retentionStrategyConfig)
+                .creationDate(ZonedDateTime.now(ZoneOffset.UTC))
+                .build();
+
+        final IndexSetConfig savedConfig = indexSetService.save(indexSetConfig);
+        clusterConfigService.write(IndexSetMigrated.create());
+
+        LOG.info("Successfully migrated index settings (database ID <{}>)", savedConfig.id());
+    }
+
+    private RetentionStrategyConfig getRetentionStrategyConfig(IndexManagementConfig indexManagementConfig) {
+        final String strategyName = indexManagementConfig.retentionStrategy();
+        final Provider<RetentionStrategy> provider = retentionStrategies.get(strategyName);
+        checkState(provider != null, "Couldn't retrieve retention strategy provider for <" + strategyName + ">");
+        final RetentionStrategy strategy = provider.get();
+        @SuppressWarnings("unchecked")
+        final Class<RetentionStrategyConfig> configClass = (Class<RetentionStrategyConfig>) strategy.configurationClass();
+        final RetentionStrategyConfig config = clusterConfigService.get(configClass);
+        return firstNonNull(config, strategy.defaultConfiguration());
```

Comment:
True.

I think we agreed at some point to actively support direct upgrades from the last minor version only (so this would be from Graylog 2.1.x in this case), but I can't find anything written on that decision. 😔